File: 20221102/모범답안/20221102_05/MatrixFullWith.js
/**
 * @param matrix {boolean[][]}
 * @param b {boolean}
 * @returns {boolean}
 */
function solution(matrix, b) {
  const n = matrix.length

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (matrix[i][j] !== b) {
        return false
      }
    }
  }

  return true
}

export default solution

File: 20221102/모범답안/20221102_05/MatrixFullWith.py
def solution(matrix, b):
    """
    :param matrix: boolean[][]
    :param b: boolean
    :return: boolean
    """

    n = len(matrix)
    for i in range(n):
        for j in range(n):
            if matrix[i][j] != b:
                return False

    return True
